### ISSUE TITLE
[Feature]: Capture temporary HTTP error calls and fall back to cache

### ISSUE THREAD

**Issue by timwsuqld** [Unleash/unleash-client-symfony#23]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Describe the new feature

Capture temporary errors in HTTP requests and fallback to cached value (without crashing everything)

### Is your feature request related to a problem? (optional)

Our GitLab instance was updated, during the update production sites would 500 error due to the 502 response they got back when checking for a feature being enabled. As this is a temporary failure, it would be good if the caching would continue to return stale data. Maybe we need 2 TTLs, the first being how soon before we try to check if the feature is enabled again (the current 30s default) and a longer one that allows us to return stale data before we error, i.e. 5 minutes.

### Describe alternatives you've considered (optional)

Everywhere I use `->isEnabled()` I have to catch the `Unleash\Client\Exception\HttpResponseException` exception and code a default for when there is an upstream error (which could be the wrong thing as I can't access the cache at that stage). This also won't work in templates etc.

### Additional context (optional)

```
Got invalid response code when getting features: 502#0 /var/www/vendor/unleash/client/src/Repository/DefaultUnleashRepository.php(49): Unleash\Client\Repository\DefaultUnleashRepository->getFeatures()
#1 /var/www/vendor/unleash/client/src/DefaultUnleash.php(61): Unleash\Client\Repository\DefaultUnleashRepository->findFeature()
#2 /var/www/vendor/unleash/symfony-client-bundle/src/Unleash/UnleashDecorator.php(44): Unleash\Client\DefaultUnleash->isEnabled()
```

**Comment by andreas-unleash** [comment]
HI, thanks for bringing this to our attention. we will discuss it internally with the team and get back to you.

**Comment by thomasheartman** [comment]
Hey, @timwsuqld 🙋 Just a little update on this: our internal team have discussed this with @RikudouSage. If I'm not mistaken, this seems like a sensible path forward and is something we'll have a look at. As the SDK maintainer, I'll let @RikudouSage take it from here.

Thanks again for the feature request! 🙌

**Comment by RikudouSage** [comment]
Hello @timwsuqld, just created first part of the PR to solve this, you might check if it works for you here: https://github.com/Unleash/unleash-client-php/pull/97

**Comment by RikudouSage** [comment]
Released in `0.9.10`.
